src/core/service.c

```diff
@@ -1832,6 +1832,13 @@ static int service_spawn_internal(
                 }
         }
 
+        if (s->restart_mode == SERVICE_RESTART_MODE_DEBUG && UNIT(s)->debug_invocation) {
```

Comment:
This can be generalized too, i.e. move to exec-invoke and check ExecParams.debug_invocation?

Reply:
This is where we set all our env vars, it's better to keep them together rather than scatter them around